actor/actionerror/duplicate_service_error.go

```diff
@@ -0,0 +1,11 @@
+package actionerror
+
+import "fmt"
+
+type DuplicateServiceError struct {
+	Name string
+}
+
+func (e DuplicateServiceError) Error() string {
```

Comment:
Am I understanding correctly that a command that would have previously succeeded will now throw an error? (in the case where a user currently has two brokers providing the same service) 

I know we've talked about this before, but I want to be really explicit about this. If we decide to make a breaking change, I think we should call out why we think it's the right decision to make.

Reply:
Yes it is a breaking change, but we don't completely understand what `call out` means.
Do you think we should send a mail to `cf-dev`, or discuss it more with the cli team, or maybe somehow add it to the release notes?

Reply:
I guess I mean I would've expected it to be mentioned in the PR description with a discussion of the reasoning behind the decision. But here is fine too. I think whenever we make a breaking change, we should be aware of what the impact will be to people we break and why it's worth doing anyway. I don't want any breaks to slip through without the PM (for example) being aware of it.

Any write up you provide could be integrated into our release notes.

Reply:
Hi @laurzca @nmaslarski I'm not aware of (or may have missed) a breaking change to support this feature. Please could you point out where this was communicated to us in the feature narrative so that I can track?

Reply:
Hey friends @abbyachau @jenspinney:

1.) We don't think this is a breaking change. Before this feature, it was not possible to register two service brokers exposing services with the same name, so it is very unlikely that a user would have encountered this scenario. 
2.) In the very unlikely situation that they did register two brokers exposing services with the same service name, then `cf create-service` would actually provision a service using seemingly random behavior: users are not asked which service from _which broker_ to provision. The user would struggle to tell which one of the services they had actually provisioned. 
3.) Without this commit, users are going to end up encountering the situation in point 2 above, which defeats the purpose of allowing them to register the same service name from two different brokers. If we view this as one track of work, then we are not introducing a breaking change.

Reply:
Hi @laurzca and other SAPIans,

I'm numbering my responses to line up with yours :)
1) I don't think it's so unlikely that there are Cloud Foundries in use now where people have registered different brokers with the same service name. This feature was released in CAPI [1.71](https://github.com/cloudfoundry/capi-release/releases/tag/1.71.0) (cf deployment [5.4.0](https://github.com/cloudfoundry/cf-deployment/releases/tag/v5.4.0)). People may have done this without even realizing it (this is especially true when space-scoped services get involved). Even if a breaking change isn't very likely, I think it's better to say "This is a breaking change, but we don't think the pain will be very bad" or "the risk of people hitting it is low" than to say it isn't a breaking change.

2) I wouldn't characterize it as "very unlikely" that people today have multiple brokers providing services with the same name (it's so easy to do, people may not realize they're doing it). But I agree this breaking change for `cf enable-service-access` is very unlikely, particulary because it's a command that isn't used very often. And yes, the existing behavior with the randomness is very strange and undesirable. But fixing that underiable behavior is a breaking change. Imagine someone has a script running in CI that for some (odd?) reason runs `cf enable-service-access`, does some provisioning and other one-off stuff, then calls `cf disable-service-access` at the end of the script. Imagine that outside of CI, someone has now created a broker with a service of the same name. The script may continue to work exactly as before if they get "lucky" with the way the CC database is ordering the service brokers. The script happily continues in CI, with no apparent difference. This PR will cause that script to now fail.

3)  "If we view this as one track of work, then we are not introducing a breaking change." I think it doesn't really matter how we view it; I think what matters for breaking change is whether people will actually feel the pain or not. If someone's day is ruined because of a breaking change, it doesn't really make them feel better if we say it was all one track of work. My point here is just that we should consider the amount of pain someone will feel if they are affected by this and weigh it against the chance of that occuring as well as the benefits of making the change.

Ultimately what I'm really driving at with all this fuss is two things: 1) It's a lot more user-friendly if we can include information about this in the CLI release notes so folks aren't totally lost if they get hit by this and 2) for the future we can often avoid introducing any kind of breaking change by considering the order that we change things in, and I think we have an opportunity to learn from this.

Reply:
I agree that running into this breaking change is unlikely, and I think we should definltey do it. (What I was disagreeing with above was whether it's likely or not that people today have service brokers with the same name; I do think that's likely.)

For the ordering of work -- if we had first changed the CLI to return an error if more than one matching service is found (instead of just picking the first one) before the loosening was done on the CC, then I think we could have done this without introducing a breaking change.

Reply:
Hi all, we can't introduce a breaking change to the CLI without bumping to a major version, which we don't plan on doing just to support this feature. I don't have the full picture, it might help if we mapped out the CLI user workflow but it'll be good to discuss a few different options on how to move forward.